packages/plugin-ext/src/plugin/command-registry.ts

```diff
@@ -100,6 +100,7 @@ export class CommandRegistryImpl implements CommandRegistryExt {
 
     // tslint:disable:no-any
     executeCommand<T>(id: string, ...args: any[]): PromiseLike<T | undefined> {
+        args = args.map(arg => this.argumentProcessors.reduce((r, p) => p.processArgument(r), arg));
```

Comment:
We already do it in `this.executeLocalCommand`, so we will run it twice.

I don't understand why we will run it for external commands?

An idea behind argument processors is to avoid sending data which is already in the plugin host for local commands not to transform something for the main side.

Reply:
Yeah, you're right, overriding service might be better solution in this case. I've updated code in following PR on che-theia side: https://github.com/eclipse/che-theia/pull/562/files.

The main problem I'm trying to solve is that, in our infrastructure each plugin runs in separate container. For example, there is the kubernetes plugin, which provides command to see the diff for the selected resource (opened in the editor). Right part for the diff takes directly from the opened editor. But left part - there is a resource stored to temporary file (in /tmp directory) in container, where plugin hosts. So we're having following situation. Kubernetes plugin calls `vscode.diff` command with two arguments: 1) path to the file, opened in the editor, everything is fine; 2) path to the file which doesn't exists in theia container, but stored in different container in /tmp directory. So we need to modify the file scheme for second argument. And with custom file content provider open this file in the diff viewer.  Smth like that.

But I've tried to override the call of `executeCommand` method and that works for the che-theia use case. So this PR can be safely closed.

Reply:
@vzhukovskii ok, glad that you find a way to resolve the issue!